run.py

```diff
@@ -137,21 +141,36 @@ def gen(test_list, simulator, simulator_yaml, output_dir, sim_only,
     sim_cmd = re.sub("<sim_opts>", sim_opts, sim_cmd)
     logging.info("Running RISC-V instruction generator")
     for test in test_list:
-      if test['iterations'] > 0:
-        rand_seed = get_seed(seed)
-        cmd = lsf_cmd + " " + sim_cmd.rstrip() + \
-              (" +UVM_TESTNAME=%s " % test['gen_test']) + \
-              (" +num_of_tests=%d " % test['iterations']) + \
-              (" +asm_file_name=%s/asm_tests/%s " % (output_dir, test['test'])) + \
-              (" -l %s/sim_%s.log " % (output_dir, test['test']))
-        cmd = re.sub("<seed>", str(rand_seed), cmd)
-        if "gen_opts" in test:
-          cmd += test['gen_opts']
-        logging.info("Generating %d %s" % (test['iterations'], test['test']))
+      """
+      If we are running a CSR test, need to call a separate python script
+      to generate directed CSR test code, located at scripts/gen_csr_test.py.
+      """
+      if test['test'] == 'riscv_csr_test' and test['iterations'] > 0:
```

Comment:
In my opinion, I think you should change the flow `if` condition. I mean:
```
if test[`iteration`] > 0:
     if test[`test`] == `riscv_csr_test`:
          ....
          ....
     else:
          ....
          ....
     if lsf_cmd:
          cmd_list.append(cmd)
     else:
          run_cmd(cmd, timeout_s)
```

What do you think?

Reply:
yes, you're right, that would be more streamlined logic.